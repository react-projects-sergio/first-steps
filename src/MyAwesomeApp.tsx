const fisrtName = "Sergio";
const lastName = "Barreras";

const favoriteGames = ["Elden Ring", "Resident Evil"];
const isActive = false;

const addres = {
  zipCode: "ABC-123",
  country: "Mexico",
};

export function MyAwesomeApp() {
  return (
    <>
      <h1>{fisrtName}</h1>
      <h3>{lastName}</h3>

      <p>{favoriteGames.join(", ")}</p>
      <p>{2 + 2}</p>

      <h2>{isActive ? "Activo" : "Inactivo"}</h2>

      <p>{JSON.stringify(addres)}</p>
    </>
  );
}
