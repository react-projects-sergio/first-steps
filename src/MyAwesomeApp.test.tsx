import { describe, test, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { MyAwesomeApp } from "./MyAwesomeApp";

describe("MyAwesomeApp", () => {
  test("should render firstName and lastName - render", () => {
    const { container } = render(<MyAwesomeApp />);

    const h1 = container.querySelector("h1");
    const h3 = container.querySelector("h3");

    expect(h1?.innerHTML).toContain("Sergio");
    expect(h3?.innerHTML).toContain("Barreras");
  });

  test("should render firstName and lastName - screen", () => {
    render(<MyAwesomeApp />);
    screen.debug();

    const h1 = screen.getByRole("heading", { level: 1 });
    const h3 = screen.getByRole("heading", { level: 3 });

    console.log(h1.innerHTML);
    expect(h1?.innerHTML).toContain("Sergio");
    expect(h3?.innerHTML).toContain("Barreras");
  });

  test("should match snapshot", () => {
    const { container } = render(<MyAwesomeApp />);

    expect(container).toMatchSnapshot();
  });
});
