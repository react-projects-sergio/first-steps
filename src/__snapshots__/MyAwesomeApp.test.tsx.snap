// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <h1>
    Sergio
  </h1>
  <h3>
    Barreras
  </h3>
  <p>
    Elden Ring, Resident Evil
  </p>
  <p>
    4
  </p>
  <h2>
    Activo
  </h2>
  <p
    style="background-color: rgb(211, 211, 211); border-radius: 10px; padding: 10px;"
  >
    {"zipCode":"ABC-123","country":"Mexico"}
  </p>
</div>
`;
